backend/inspirehep/pidstore/minters/bai.py

```diff
@@ -4,14 +4,76 @@
 #
 # inspirehep is free software; you can redistribute it and/or modify it under
 # the terms of the MIT License; see LICENSE file for more details.
-
+from flask import current_app
 from inspire_utils.record import get_values_for_schema
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
 
 from inspirehep.pidstore.minters.base import Minter
+from inspirehep.pidstore.providers.bai import InspireBAIProvider
 
 
 class BAIMinter(Minter):
     pid_type = "bai"
+    provider = InspireBAIProvider
+
+    @classmethod
+    def mint(cls, object_uuid, data):
+        minter = cls(object_uuid, data)
+        minter.validate()
+        pid_values = minter.get_pid_values()
+        if pid_values:
+            for pid_value in pid_values:
+                minter.create(pid_value)
+        else:
+            if current_app.config.get("FEATURE_FLAG_ENABLE_BAI_PROVIDER", False):
```

Comment:
this is the only place you put a feature flag. Are `create` and `update` calling this method somehow?

Reply:
I'm updating now whole BaiMinter and parts of BaseMinter as methods are quite similar but few modifications are required. It should simplify a little Minters. 
Still there will be problem with feature flag on BaiMinter as we had BaiMinter before but with different Provider and we cannot easily change it based on feature flag :/ I could just create `OldBaiMinter` But this solution is quite ugly.